### PR TITLE
Customizable annotations and labels on CRD

### PR DESCRIPTION
**Pull request by andreaTP**
## Description
Add customizable annotations and labels for CRDs

## Type of change
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [x] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/decorator/ResourceProvidingDecorator.java [hunk 1]
```diff
@@ -15,6 +15,24 @@
...
 
+  protected Map<String, String> toMap(String[] arr) {
+    Map<String, String> res = new HashMap<>();
+    if (arr != null) {
+      for (String e : arr) {
+        String[] splitted = e.split("\\=");
+        if (splitted.length == 2) {
+          res.put(splitted[0], splitted[1]);
+        } else {
+          throw new IllegalArgumentException("Invalid value: " + e + " cannot be parsed as a Map");
```
Comment (metacosm) [inline crd-generator/api/src/main/java/io/fabric8/crd/generator/decorator/ResourceProvidingDecorator.java:31]:
```suggestion
          throw new IllegalArgumentException("Invalid value: " + e + " cannot be parsed as a key-value pair. Expected format is 'key=value'.");
```

File: generator-annotations/src/main/java/io/fabric8/generator/annotation/Annotations.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+package io.fabric8.generator.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Annotations {
```
Comment (metacosm) [inline generator-annotations/src/main/java/io/fabric8/generator/annotation/Annotations.java:25]: Would this and `Labels` annotations be used by the Java generator as well? Seems that, at least right now, they'd make more sense if they were in the CRD generator module/package.

Comment (andreaTP) [reply generator-annotations/src/main/java/io/fabric8/generator/annotation/Annotations.java:25]: sorry, I missed your review @metacosm , I agree I can move those annotations

Comment (andreaTP) [reply generator-annotations/src/main/java/io/fabric8/generator/annotation/Annotations.java]: done

**Comment by andreaTP** [comment]
@metacosm this should be ready for another review round 🙂

**Comment by andreaTP** [comment]
this is not necessary but it would be a nice-to-have for 6.2 @manusa

**Review by metacosm — APPROVED** [review]
LGTM, thank you!

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/decorator/ResourceProvidingDecorator.java [hunk 3]
```diff
@@ -15,6 +15,25 @@
...
+        if (splitted.length == 2) {
+          res.put(splitted[0], splitted[1]);
+        } else {
+          throw new IllegalArgumentException(
+              "Invalid value: " + e + " cannot be parsed as a key-value pair. Expected format is 'key=value'.");
+        }
+      }
+    }
+    return res;
+  }
```
Comment (manusa) [inline crd-generator/api/src/main/java/io/fabric8/crd/generator/decorator/ResourceProvidingDecorator.java:37]: I'm not sure about the implications of this method and the opposite `CustomResourceInfo.toStringArray`

What would happen if there's a label or annotation similar to:

```java
String [] test = new String[] {
  "edge.example.com/math= 1 + 1 = 2",
  "edge.example.com/annoying= =======> () <======"
};

// Or for the opposite method
Map.of(
  "edge.example.com/math", "1 + 1 = 2",
  "edge.example.com/annoying", "=======> () <======"
};

```

What are the implications? is this a possible scenario?

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**